setup/scripts/automail.php

```diff
@@ -71,10 +71,13 @@ function_exists('curl_version') or die('CURL support required');
             $code = 69;
             break;
         case 500: //Server error.
-        default: //Temp (unknown) failure - retry 
+        default: //Temp (unknown) failure - retry
             $code = 75;
     }
 }
 
+if (preg_match('/HTTPS Protocol Required/', $result, $m) == 1) {
```

Comment:
Curious why you didn't rely on http code returned here! Asking because we might end up translating response text down the road which will break the check here of course.

Reply:
Because 400 is super generic and not related to Force HTTPS specifically. I can return a specific response code and check against that but if we return the same response code for something else the check will be inaccurate. Maybe I can set a specific response header and check that instead?

Reply:
As long as you're not using a code like 403, 404 etc that server might actually return in normal operations then I think you're fine in the context of osTicket specific handshake. 

We use 201 for example to indicate a ticket was created. You could use  495 for example... which ngnix uses in the context of invalid ssl certificate which I think is not an issue within this specific use case. 

Just be sure to document it in `class.http.php` -  as you can see 400 is already in there.

```php
class Http {

    function header_code_verbose($code) {
        switch($code):
        case 200: return '200 OK';
        case 201: return '201 Created';
        case 204: return '204 No Content';
        case 205: return '205 Reset Content';
        case 306: return '306 Switch Proxy';
        case 400: return '400 Bad Request';
        case 401: return '401 Unauthorized';
        case 403: return '403 Forbidden';
        case 404: return '404 Not Found';
        case 405: return '405 Method Not Allowed';
        case 416: return '416 Requested Range Not Satisfiable';
        case 418: return "418 I'm a teapot";
        case 422: return '422 Unprocessable Entity';
        default:  return '500 Internal Server Error';
        endswitch;
    }

```

Reply:
That’s a bet